core/db/migrate/20150113002122_create_spree_promotion_codes.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreePromotionCodes < ActiveRecord::Migration
+  def change
+    create_table :spree_promotion_codes do |t|
+      t.references :promotion, index: true
+      t.string :value
+      t.datetime :starts_at
```

Comment:
I can see both sides here.  I'm not sure which I'd prefer, but my initial thought was that this would work as @athal7 expected. It seems to me like bulk updating codes would be more important than reducing promotion duplication.